### PR TITLE
Features/rbac

### LINKED ISSUE #2505: Markdown editor does not support split line ( divider / dividing line / <hr> )
**Informations**
- **Node.js version**: 10.14.1
- **NPM version**: 6.4.1
- **Strapi version**: 3.0.0-alpha.16
- **Database**: PostgreSQL 10
- **Operating system**: Windows 10

**What is the current behavior?**

image

**Steps to reproduce the problem**

Copy and paste:
```
## Hyphens

---


## Asterisks

***


## Underscores
___
```

**What is the expected behavior?**

image

image

**Suggested solutions**

### LINKED ISSUE #5111: Performance issues on bigger forms
**Describe the bug**
I'm using Strapi as a CMS to control the content of my website. I wanted to be able to control every bigger element on my page via Strapi, so I've created this pretty flexible content-type (which is also pretty big). Unfortunately, when the Dynamic Zone field has already ~10 entries, the form becomes totally unusable (there's a massive lag on input).

**Steps to reproduce the behavior**
1. Clone https://github.com/sarneeh/strapi-strapi-issues-5111
2. Run Strapi `npm i && npm run develop`
2. Go to Pages -> "Test" entry (the database is included)
3. Try to type anything into a text field

Login credentials: `test/test1234`

**Expected behavior**
I expect the form to work fast.

**Screenshots**
strapi-issue

**System**
- Node.js version: 12
- NPM version: 6
- Strapi version: beta.18.6
- Database: SQLite
- Operating system: MacOS

### LINKED ISSUE #6051: Nested lists in Rich text field (markdown) doesn't work
**Describe the bug**

Nested lists in Rich text field (markdown) doesn't work

**Steps to reproduce the behavior**

1. Create a rich text field
2. Create a nested list
3. The markdown preview show just one list, without nested

Screenshot 2020-05-04 at 18 26 37

**Expected behavior**

Show the nested lists:

Screenshot 2020-05-05 at 09 53 22

**System**

- Node.js version: v12
- NPM version: v6
- Strapi version: v3.0.0-beta.20.1
- Database: MongoDB
- Operating system: macOS

### PR DESCRIPTION
**Pull request by alexandrebodin**
#### Description of what you did:

Add RBAC feature

Fix #6051 
Fix #5111
Fix #2505
Fix https://github.com/strapi/strapi/issues/3194

### CODE REVIEW

File: packages/strapi-connector-bookshelf/lib/buildQuery.js [hunk 1]
```diff
@@ -135,71 +134,80 @@ const buildJoinsAndFilter = (qb, model, whereClauses) => {
...
+    model,
+    joins: {},
+  };
+
   /**
-   * Builds a Strapi query tree easy
-   * @param {Array<Object>} whereClauses - Array of Strapi where clause
-   * @param {Object} model - Strapi model
-   * @param {Object} queryTree - queryTree
+   * Returns the SQL path for a qery field.
```
Comment (derrickmehaffy) [inline packages/strapi-connector-bookshelf/lib/buildQuery.js:150]: query field

File: packages/strapi-connector-mongoose/lib/mount-models.js [hunk 2]
```diff
@@ -27,6 +27,18 @@ module.exports = ({ models, target }, ctx) => {
       primaryKeyType: 'string',
     });
 
+    if (!definition.uid.startsWith('strapi::') && definition.modelType !== 'component') {
+      definition.attributes['created_by'] = {
```
Comment (derrickmehaffy) [inline packages/strapi-connector-mongoose/lib/mount-models.js:31]: Should these be using underscores? The `updatedAt` and `createdAt` are what is normally used on mongoose so should we also use `createdBy` and `updatedBy` for mongoose and keep the underscores for SQL? (Honestly not sure why the two are different for different ORMs but 🤷 )

Comment (alexandrebodin) [reply packages/strapi-connector-mongoose/lib/mount-models.js:31]: For simplicity's sake we are using one unique name everywhere for this.

File: packages/strapi-plugin-content-manager/admin/src/utils/generatePermissionsObject.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const addSubjectToPermissionsArray = (array, uid) => array.map(data => ({ ...data, subject: uid }));
+
+const generatePermissionsObject = uid => {
+  const permissions = {
+    create: [{ action: 'plugins::content-manager.explorer.create', subject: null }],
+    delete: [{ action: 'plugins::content-manager.explorer.delete', subject: null }],
+    read: [
+      { action: 'plugins::content-manager.explorer.read', subject: null },
+      // { action: 'plugins::content-manager.explorer.update', subject: null },
```
Comment (soupette) [inline packages/strapi-plugin-content-manager/admin/src/utils/generatePermissionsObject.js:9]: To be removed

File: packages/strapi-connector-mongoose/lib/mount-models.js (hunk 2 above)
Comment (derrickmehaffy) [reply packages/strapi-connector-mongoose/lib/mount-models.js:31]: Good to know, wondering if in the future we should look at changing the mongoose timestamp fields to match 🤔

**Comment by sebacampos** [comment]
Amazing work! 💪 I’m really looking forward to have this PR merged and `3.1.0` released 🤩🙌🚀 

Just to clear up any doubts:  RBAC feature (3 admin panel roles) will be available for CE right?

image

**Pull request merged**